src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy

```diff
@@ -167,6 +172,14 @@ enum RequisitionStatus {
         ]
     }
 
+    static String getRequestStatusLabel(RequisitionStatus requisitionStatus, ShipmentStatus shipmentStatus) {
```

Comment:
I think you should not mix together RequisitionStatus and ShipmentStatus here in the RequisitionStatus enum. I feel like it should rather be in StockMovementService instead of here. Plus I think you should do it a bit similarly to the way it's done in the FormatTagLib - long story short - take the `getClass().getSimpleName()`, and then pass it to `g.message`. Just decide which status to use, and return:
 `"${g.message(code: 'enum.' + status.getClass().getSimpleName() + '.' + status.name())}"`

Reply:
Yeah, I agree with @awalkowiak. If nothing else the logic should probably be in the StockMovement command. You probably need to add more states to the StockMovementStatusCode enum and map them in the fromStockMovementStatus and toStockMovementStatus methods.